### PR TITLE
adding docstrings to asg.py

### PR DESCRIPTION
**Pull request by joshuaroot**
* adding docstrings to asg.py

### CODE REVIEW

File: c7n/resources/asg.py [hunk 1]
```diff
@@ -94,6 +94,7 @@ def initialize(self, asgs):
 @filters.register('security-group')
 class SecurityGroupFilter(
         net_filters.SecurityGroupFilter, LaunchConfigFilterBase):
+    """Filters ASG based on security group"""
```
Comment (kapilt) [inline c7n/resources/asg.py]: these need documentation once on the base class, not on every subclass

File: c7n/resources/asg.py [hunk 2]
```diff
@@ -111,6 +112,7 @@ def process(self, asgs, event=None):
 
 @filters.register('subnet')
 class SubnetFilter(net_filters.SubnetFilter):
+    """Filters ASG based on associated subnet"""
```
Comment (kapilt) [inline c7n/resources/asg.py]: these need documentation once on the base class, not on every subclass

File: c7n/resources/asg.py [hunk 3]
```diff
@@ -457,6 +516,19 @@ def get_resource_date(self, i):
 
 @filters.register('vpc-id')
 class VpcIdFilter(ValueFilter):
+    """Filters ASG based on the VpcId
+
```
Comment (kapilt) [inline c7n/resources/asg.py]: this could use an explanation on why a simple value filter would not work.

File: c7n/resources/asg.py [hunk 4]
```diff
@@ -521,6 +624,24 @@ def process(self, asgs, event=None):
 
 @actions.register('resize')
 class Resize(BaseAction):
+    """Action to resize the min/max instances in an ASG
+
+    Resizing of scaling groups desired/minimum size is limited to the
```
Comment (kapilt) [inline c7n/resources/asg.py]: the limitations on this needs  a **note** prefix in bold

File: c7n/resources/asg.py [hunk 5]
```diff
@@ -595,6 +731,23 @@ def process_asg_set(self, asgs, key):
 @actions.register('tag')
 @actions.register('mark')
 class Tag(BaseAction):
+    """Action to add a tag to an ASG
+
```
Comment (kapilt) [inline c7n/resources/asg.py]: explain what propagate does

File: c7n/resources/asg.py [hunk 6]
```diff
@@ -1027,6 +1305,20 @@ def __call__(self, config):
 
 @LaunchConfig.action_registry.register('delete')
 class LaunchConfigDelete(BaseAction):
+    """Filters all launch configurations that are current not being used
```
Comment (kapilt) [inline c7n/resources/asg.py]: this sentence needs rewording for grammar.

**Comment by joshuaroot** [comment]
docstring examples have been tested & validated

**Pull request merged**